vector/src/main/java/im/vector/app/features/home/room/detail/composer/MessageComposerViewModel.kt

```diff
@@ -741,13 +749,35 @@ class MessageComposerViewModel @AssistedInject constructor(
     }
 
     private fun handleEndAllVoiceActions(deleteRecord: Boolean) {
+        voiceMessageHelper.clearTracker()
         voiceMessageHelper.stopAllVoiceActions(deleteRecord)
     }
 
+    private fun handleInitializeVoiceRecorder(attachmentData: ContentAttachmentData) {
+        voiceMessageHelper.initializeRecorder(attachmentData)
+        setState { copy(voiceRecordingUiState = VoiceMessageRecorderView.RecordingUiState.Draft) }
+    }
+
     private fun handlePauseRecordingVoiceMessage() {
         voiceMessageHelper.pauseRecording()
     }
 
+    private fun handleEntersBackground(composerText: String) {
+        val isVoiceRecording = com.airbnb.mvrx.withState(this) { it.isVoiceRecording }
```

Comment:
we want the synchronous state reading here as we need to act during `onPause` and clear any view state (clearTracker) whilst on the main thread

if we do this async then we can end up removing the waveforms due to `clearTracker` triggering after the reinitialising the views